.github/workflows/e2e-kind.yml

```diff
@@ -22,5 +22,5 @@ jobs:
       run: |
         kind create cluster
         kind export kubeconfig
-    - name: Run e2e tests
```

Comment:
These aren't really running e2e tests so much as they're ensuring we can deploy OLM to a specific target; how valuable is this?

Reply:
I had previously attempted to remove these checks entirely but there was some pushback in #olm-dev as they're mainly in place to ensure there's no regression when running OLM on various cluster environments.

Reply:
But for context, I had renamed this check and simply misjudged what these checks were trying to accomplish when splitting up that e2e-tests GHA workflow into their own workflows.